src/components/Fuel/FuelForm.tsx

```diff
@@ -0,0 +1,118 @@
+    }
+
+    /* Function that updates the state */
+    function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+        /* Destructuring the event.target object */
+        const {name, value, valueAsNumber} = event?.target;
+        setfuelData(prevFuelData => {
+            return {
+                ...prevFuelData,
+                [name]: name == "fuel_source_value" ? valueAsNumber : value
+            }
+        })  
+    }
+
+    /* Renders the units conditionally */
+    const unitOptions = fuelData.fuel_source_type == "ng" ? ["thousand_cubic_feet", "btu"] : ["gallon", "btu"];
+    
+    function resetForm() {
+        setfuelData(
+            {
+                fuel_source_type: "ng",
+                fuel_source_unit: "btu",
+                fuel_source_value: 0
+            }
+        )
+    }
+
+    
+    return (
+        <div>
+                          
+                <Label htmlFor="fuel_source_type">Which fuel do you use?</Label>
+                <Select 
+                    id="fuel_source_type"
+                    value={fuelData.fuel_source_type}
+                    onChange={handleChange}
+                    name="fuel_source_type"
+                >
+                    <option value="">--- Choose</option>
+                    <option value="ng">Natural Gas</option>
+                    <option value="dfo">Home Heating and Diesel Fuel</option>
+                    <option value="pg">Propane Gas</option>
+                    <option value="ker">Kerosene</option>
+                </Select>
+
+                <br/>
+
+                <Label htmlFor="unit">Which measuring unit do you use?</Label>
+                <Select
+                    id="fuel_source_unit"
+                    value={fuelData.fuel_source_unit}
+                    onChange={handleChange}
+                    name="fuel_source_unit"
+                >
+                    <option value="">-- Choose</option>
+                    <option value={unitOptions[0]}>{unitOptions[0]}</option>
+                    <option value={unitOptions[1]}>{unitOptions[1]}</option>
+                </Select>
+
+                <br/>
+
+                <Label htmlFor="value">What is the quantity of fuel you consumed? (based on the unit you&apos;ve chosen above)</Label>
+                <TextInput
+                id="fuel_source_value"
+                type="number"
+                value={fuelData.fuel_source_value}
+                onChange={handleChange}
+                name="fuel_source_value"
+              />  
+
+                <div>
+                    <Button
+                        color="light"
+                        onClick={resetForm}
+                    >Reset</Button>
+                    
+                    <Button
+                        type="submit"
+                        disabled={!fuelData.fuel_source_type || !fuelData.fuel_source_unit || fuelData.fuel_source_value<1 ? true : false}
+                        color="default"
```

Comment:
Can you change this color to success so it matches the styling of the other calculators?

Reply:
I was in doubt about the colors... Thanks for clearing it up :)

Reply:
Done!